Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2027,6 +2030,10 @@ extension BrowserViewController: TabManagerDelegate {
             topTabsViewController?.updateTabCount(count, animated: animated)
         }
     }
+
+    @objc func tabManagerUpdateCount() {
+        updateTabCountUsingTabManager(self.tabManager)
```

Comment:
This can be further improved and moved to the background thread as only parts where we update UI needs to be updated. For now, keeping it in but if we get more complaints of slowness we should move it to background thread.

Note: computed properties like private / normal tabs also cause issues when being accessed all the times so another improvement could be to change that and not have those properties computed.